provider/dynamodb.go

```diff
@@ -169,181 +218,172 @@ func (store *dynamodbOnlineStore) UpdateMetadataTable(tablename string, valueTyp
 	return nil
 }
 
-func (store *dynamodbOnlineStore) GetFromMetadataTable(tablename string) (ValueType, error) {
+func (store *dynamodbOnlineStore) getFromMetadataTable(tablename string) (*dynamodbTableMetadata, error) {
 	input := &dynamodb.GetItemInput{
 		TableName: aws.String("Metadata"),
-		Key: map[string]*dynamodb.AttributeValue{
-			"Tablename": {
-				S: aws.String(tablename),
+		Key: map[string]types.AttributeValue{
+			"Tablename": &types.AttributeValueMemberS{
+				Value: tablename,
 			},
 		},
 	}
-	output_val, err := store.client.GetItem(input)
+	output_val, err := store.client.GetItem(context.TODO(), input)
 	if len(output_val.Item) == 0 {
-		return NilType, fferr.NewDatasetNotFoundError("", "", fmt.Errorf("table %s not found", tablename))
+		return nil, fferr.NewDatasetNotFoundError("", "", fmt.Errorf("table %s not found", tablename))
 	}
 	if err != nil {
 		wrappedErr := fferr.NewExecutionError(pt.DynamoDBOnline.String(), err)
 		wrappedErr.AddDetail("tablename", tablename)
-		return NilType, wrappedErr
+		return nil, wrappedErr
 	}
-	metadata_item := Metadata{}
-	err = dynamodbattribute.UnmarshalMap(output_val.Item, &metadata_item)
-	if err != nil {
+	var entry dynamodbMetadataEntry
+	if err = attributevalue.UnmarshalMap(output_val.Item, &entry); err != nil {
 		wrappedErr := fferr.NewExecutionError(pt.DynamoDBOnline.String(), err)
 		wrappedErr.AddDetail("tablename", tablename)
-		return NilType, wrappedErr
+		return nil, wrappedErr
 	}
-	return ScalarType(metadata_item.Valuetype), nil
+	tableMeta, err := entry.ToTableMetadata()
+	if err != nil {
+		return nil, fferr.NewInternalError(err)
+	}
+	return tableMeta, nil
 }
 
-func GetTablename(prefix, feature, variant string) string {
+func formatDynamoTableName(prefix, feature, variant string) string {
```

Comment:
just a thought, when dealing w/ clickhouse it was SUPER annoying figure out when table name needs to be formatted vs not. I wonder if it makes sense to introduce a type FormattedTableName so that it's passed intentionally.

Reply:
In a different PR, I'll solve this